spec/mailers/claim_mailer_spec.rb

```diff
@@ -32,4 +32,20 @@
       expect(mail.body.encoded).to match("Email studentloanteacherpayment@digital.education.gov.uk giving your reference number: #{claim.reference} if you have any questions.")
     end
   end
+
+  describe "#rejected" do
+    let(:claim) { create(:claim, :submitted, first_name: "John", middle_name: "Fitzgerald", surname: "Kennedy") }
+    let(:mail) { ClaimMailer.rejected(claim) }
+
+    it "renders the headers" do
+      expect(mail.subject).to eq("Your claim to get your student loan repayments back has been rejected, reference number: #{claim.reference}")
+      expect(mail.to).to eq([claim.email_address])
+    end
+
+    it "renders the body" do
+      expect(mail.body.encoded).to match("Dear John Kennedy,")
+      expect(mail.body.encoded).to match("Unfortunately your claim to get your student loan payments back has been denied.")
+      expect(mail.body.encoded).to match("If you feel that we have made a mistake in the processing of your application or something is incorrect please email studentloanteacherpayment@digital.education.gov.uk giving your reference number: #{claim.reference} if you have any questions about your claim.")
```

Comment:
I have mixed feelings about specs that precisely verify that content matches. What we gain in exactness, we lose in the robustness of the spec as it's much harder to make copy adjustments without also breaking specs. What do you think to reworking these expectations to be a bit looser such that they still give us confidence the template is rendering without error, and that it is the right template, but without coupling them so tightly to the actual content so that content can easily be tweaked without having to change the spec?

Reply:
I've got round this in the past by using translations in the emails and using the translations in the tests too, but I guess that makes the content of the emails less clear. I'd say the only thing we _really_ need to test here is for the presence of dynamic content (so, the name in this case).

Reply:
Checking dynamic content is good. But I'd say we'd also want to perform a minimum match against the body _and substance_ of the email itself, otherwise the code could be changed torender the "approved" email template and the spec would still pass.

Reply:
Yeah, it's a shame we can't check what template is rendered in a similar way to controller specs (even though I know controller specs are no longer The Preferred Way Of Doing Things)